### ISSUE TITLE
unified transformer : training loss start raising suddenly

### ISSUE THREAD

**Issue by jprobichaud** [wenet-e2e/wenet#279]
I'm reusing the librispeech configuration for the unified transformer, but I'm using our own internal data (about 10k hours), with a 10,000 BPE model as the tokenizer, on 8 A100 GPUs.

Since I had good success with the original model configuration, and as I have much more data to throw in for training, I increased the size of the model a bit.

Overall, the differences with the original config are:
* I set the batch_type to static
* I set the batch size to 8
* I set the accum_grad to 4
* I set scheduler_conf.warmup_steps to 25000
* encoder_onf.output_size = 1024 (instead of 512)
* encoder_conf.num_blocks = 8 (instead of 2)

Everything goes fine for a while, training loss is going down and suddenly, there's a big jump in the training loss (towards the end of the first epoch) and it stays high (for now at least).

![image](https://user-images.githubusercontent.com/9254379/113798046-6b4eb500-9720-11eb-85cd-fc3768b59731.png)

Is there something special I should look at to avoid this effect?

**Comment by robin1001** [comment]
how many epochs now? maybe:
1.  the sudden jump is caused by some bad data.
2. the training is not stable enough, try to use smaller learning rate.
We have tried the unified solution on tens of thousands Chinese data, it works quite well.

**Comment by chwma0** [comment]
I also encountered the problem of non-convergence using internal data (7.5k hours). The main reason is that the key corresponding to the same line in text and wav.scp does not match. After the modification, I am running the run.sh script again.

**Comment by jprobichaud** [comment]
Thanks for your quick replies.

About the data, I'm pretty confident of that data because we used it with good success with some ESPnet configuration.  I will try to lower the learning rate.  

@chwma0 : do you mean that the sorting of the rows could be wrong or that there is a true mismatch between the ids?

**Comment by whiteshirt0429** [comment]
perhaps  decrease learning rate dose help

**Comment by jprobichaud** [comment]
Indeed decreasing the learning rate helped a lot.  Thanks!

**Comment by jprobichaud** [comment]
We are evaluating various toolkits right now.  The unified conformer is really interesting for our needs as we need both batch and streaming functionalities.  OTOH, some interesting features available in other toolkits are missing in wenet.  But all in all, this toolkit is very interesting.

We might start looking at supporting GPUs in your C++ runtime.  Is that something you also have on your radar?

**Comment by chwma0** [comment]
> Indeed decreasing the learning rate helped a lot. Thanks!

Can you refer to the key parameters to solve the problem?
